src/parser/TraverseSentences.h

```diff
@@ -563,14 +564,23 @@ class YieldSentence final : public Sentence {
 
 class GroupBySentence final : public Sentence {
 public:
-    GroupBySentence() {
+    GroupBySentence(YieldClause* yield = nullptr,
+                    WhereClause* having = nullptr,
+                    GroupClause* groupby = nullptr) {
         kind_ = Kind::kGroupBy;
+        yieldClause_.reset(yield);
+        havingClause_.reset(having);
+        groupClause_.reset(groupby);
     }
 
     void setGroupClause(GroupClause *clause) {
         groupClause_.reset(clause);
     }
 
+    void setHavingClause(WhereClause *clause) {
```

Comment:
where use it? and what's the meaning?

Reply:
Not used yet, but we'll support Having clause later.